tests/python_client/testcases/test_query.py

```diff
@@ -686,6 +686,141 @@ def test_query_expr_json_contains_pagination(self, enable_dynamic_field):
             res = collection_w.query(expression, limit=limit, offset=offset)[0]
             assert len(res) == limit - offset
 
+    @pytest.mark.tags(CaseLabel.L1)
+    def test_query_expr_empty_without_limit(self):
+        """
+        target: test query with empty expression and no limit
+        method: query empty expression without setting limit
+        expected: raise error
+        """
+        # 1. initialize with data
+        collection_w = self.init_collection_general(prefix, True)[0]
+
+        # 2. query with no limit and no offset
+        error = {ct.err_code: 1, ct.err_msg: "empty expression should be used with limit"}
+        collection_w.query("", check_task=CheckTasks.err_res, check_items=error)
+
+        # 3. query with offset but no limit
+        collection_w.query("", offset=1, check_task=CheckTasks.err_res, check_items=error)
```

Comment:
what about query() or query(offset), need a case here.

Reply:
Added in test_query_empty.  
for query(offset), it means nothing, because offset can be effective only when limit is given too.